### ISSUE TITLE
Clang's --start-no-unused-arguments not honored, causing errors due to unused arguments

### ISSUE THREAD

**Issue by mstorsjo** [ccache/ccache#1397]
### How to reproduce ###

1. With Clang 14 or newer, run `clang --start-no-unused-arguments -stdlib=libc++ --end-no-unused-arguments -Werror -c hello.c`, note that this executes cleanly, with no printed errors or warnings.
2. Rerun the same command with `ccache` prefixed. Observe that it fails with `clang: error: argument unused during compilation: '-stdlib=libc++' [-Werror,-Wunused-command-line-argument]`.

### Actual behavior ###
Rerunning the failing command with `CCACHE_DEBUG=1` and inspecting the log shows that ccache executed this command in order to preprocess the source:
```
Executing clang --start-no-unused-arguments --end-no-unused-arguments -Werror -stdlib=libc++ -c -fcolor-diagnostics -o hello.o hello.c
```

By reordering the command line arguments, moving `-stdlib=libc++` outside of the region with `--start-no-unused-arguments` and `--end-no-unused-arguments`, clang triggers a warning about unused command line arguments, which is treated as an error here.

### Expected behavior ###
Ideally, ccache would not reorder arguments outside of the region enclosed by `--start-no-unused-arguments` and `--end-no-unused-arguments`

### Environment ###
Tested with the latest ccache from git as of today (February 6th 2024, git commit b5c103fdb1e4cc6669c3bc77617c29aaad78004c), version v4.9-41-gb5c103fd.

Tested with a current git version of Clang 18.x - any version of Clang since version 14 supports this feature for ignoring specific ranges of arguments.

**Comment by jrosdahl** [comment]
Thanks for the bug report. This has the same root cause as #738, so I'll close this as a duplicate.

I'll try to find time to fix this in ccache 4.10.

**Comment by jrosdahl** [comment]
Duplicate of #738
